daemon/cmd/fqdn.go

```diff
@@ -188,7 +188,10 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState, preCache
 			namesToClean = append(namesToClean, d.dnsPoller.DNSHistory.GC(GCStart, nil)...)
 
 			// Cleanup each endpoint cache, deferring deletions via DNSZombies.
+			d.RLock()
```

Comment:
Why is this lock needed here? I have two problems with this line:

`GetEndpoints` uses `EndpointManager` mutex to RLock, which should be enough to not race.

This lock is used in actual implementation code, which is in contradiction to comment in `daemon/cmd/daemon.go` which reads "It is not used for implementation code."

Reply:
Thanks for the review. I've found a way to avoid using a mutex: bb4712c4f9ed72fef45eef8a6d6cef34cf91dab3